src/NGen.cpp

```diff
@@ -756,6 +759,8 @@ int main(int argc, char* argv[]) {
 #if NGEN_WITH_ROUTING
     if (mpi_rank == 0) { // Run t-route from single process
```

Comment:
Please either incorporate a `MPI_Gather` calls outside the conditional to properly support parallel runs, or open a 2-point story assigned to me to implement that support.